docs/user-guide/private-repositories.md

```diff
@@ -41,24 +41,32 @@ The Argo CD UI don't support configuring SSH credentials. The SSH credentials ca
 argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-key-path ~/.ssh/id_rsa
 ```
 
-## Self-Signed Certificates
+## HTTPS Self-signed Certificate
 
-If you are using self-hosted Git hosting service with the self-signed certificate then you need to disable certificate validation for that Git host.
-Following options are available:
+We do not currently have first-class support for this. See [#1513](https://github.com/argoproj/argo-cd/issues/1513).
+
+As a work-around, you can customize your Argo CD image. See [#1344](https://github.com/argoproj/argo-cd/issues/1344#issuecomment-479811810)
 
-Add repository using Argo CD CLI and `--insecure-ignore-host-key` flag:
+## Unknown SSH Hosts
 
+If you are using a self-hosted Git service over SSH then you need to disable host verification for that Git host.
+
+Following options are available:
+
+(1) Add repository using Argo CD CLI and `--insecure-ignore-host-key` flag:
 
 ```bash
-argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-key-path ~/.ssh/id_rsa
+argocd repo add git@github.com:argoproj/argocd-example-apps.git --ssh-private-key-path ~/.ssh/id_rsa --insecure-ignore-host-key 
 ```
+
+!!! warning "This does not work for Kustomize remote bases or custom plugins"
+    This does not yet work for them. For Kustomize support, see [#827](https://github.com/argoproj/argo-cd/issues/827).
```

Comment:
> This does not work for Kustomize remote bases or custom plugins" This does not yet work for them. For Kustomize support

Sentence repeats itself. How about:

This does yet not work for Kustomize remote bases or custom plugins. See [#827](https://github.com/argoproj/argo-cd/issues/827).